src/pupil_labs/invisible_lsl_relay/cli.py

```diff
@@ -30,36 +46,39 @@ def __init__(self, search_timeout):
         self.search_timeout = search_timeout
         self.n_reload = 0
 
-    async def get_user_selected_device(self):
+    async def get_device_from_list(self):
         async with Network() as network:
             print("Looking for devices in your network...\n\t", end="")
             await network.wait_for_new_device(timeout_seconds=self.search_timeout)
 
             while self.selected_device_info is None:
-                print("\n======================================")
-                print("Please select a Pupil Invisible device by index:")
-                print("\tIndex\tAddress" + (" " * 14) + "\tName")
-                for device_index, device_info in enumerate(network.devices):
-                    ip = device_info.addresses[0]
-                    port = device_info.port
-                    full_name = device_info.name
-                    name = full_name.split(":")[1]
-                    print(f"\t{device_index}\t{ip}:{port}\t{name}")
-
-                print()
-                print("To reload the list, hit enter. ")
-                print("To abort device selection, use 'ctrl+c' and hit 'enter'")
-                if self.n_reload >= 5:
-                    print(
-                        "Can't find the device you're looking for?\n"
-                        "Make sure the Companion App is connected to the same "
-                        "network and at least version v1.4.14."
-                    )
+                print_device_list(network, self.n_reload)
                 self.n_reload += 1
                 user_input = await input_async()
                 self.selected_device_info = evaluate_user_input(
                     user_input, network.devices
                 )
+        return self.selected_device_info.addresses[0], self.selected_device_info.port
+
+
+def get_user_defined_device(device_address):
+    ip_regex = r"\d*\.\d*\.\d*\.\d*:\d*"
```

Comment:
This regex enforces ipv4 addresses and disallows ipv6 addresses and/or domain names, e.g. `pi.local`. I suggest simplifying the parsing into
```py
try:
    address, port = device_address.split(":")
    port = int(port)  # raises ValueError if port cannot be transformed to an int
    if address == "":
        raise ValueError("Empty address")
except ValueError as exc:
    raise ValueError(
       'Device address could not be parsed in IP and port!\n '
        'Please provide the address in the format IP:port'
    ) from exc
```